### ISSUE TITLE
validate debugger works with incoming R 4.0.0 release

### ISSUE THREAD

**Issue by kevinushey** [rstudio/rstudio#5888]
Reminder issue; motivated by https://community.rstudio.com/t/rstudio-crashes-when-debugging/47214.

**Comment by jmcphers** [comment]
This crash is happening when we invoke the function `.rs.sourceFileFromRef`. Seems likely that RCNTXT has changed?

```
#29	0x0000000115feae3c in R_tryEval ()
#30	0x000000010df23b09 in rstudio::r::exec::(anonymous namespace)::evaluateExpressionsUnsafe(SEXPREC*, SEXPREC*, SEXPREC**, rstudio::r::sexp::Protect*, rstudio::r::exec::(anonymous namespace)::EvalType) at /Users/jmcphers/git/rstudio/src/cpp/r/RExec.cpp:154
#31	0x000000010df2517b in rstudio::r::exec::(anonymous namespace)::evaluateExpressions(SEXPREC*, SEXPREC*, SEXPREC**, rstudio::r::sexp::Protect*) at /Users/jmcphers/git/rstudio/src/cpp/r/RExec.cpp:187
#32	0x000000010df24e0e in rstudio::r::exec::RFunction::call(SEXPREC*, bool, SEXPREC**, rstudio::r::sexp::Protect*) at /Users/jmcphers/git/rstudio/src/cpp/r/RExec.cpp:422
#33	0x000000010df24fa9 in rstudio::r::exec::RFunction::call(SEXPREC*, SEXPREC**, rstudio::r::sexp::Protect*) at /Users/jmcphers/git/rstudio/src/cpp/r/RExec.cpp:384
#34	0x000000010df24f66 in rstudio::r::exec::RFunction::call(SEXPREC**, rstudio::r::sexp::Protect*) at /Users/jmcphers/git/rstudio/src/cpp/r/RExec.cpp:379
#35	0x000000010df18046 in rstudio::r::context::RCntxt::fileName(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >*) const at /Users/jmcphers/git/rstudio/src/cpp/r/RCntxt.cpp:131
#36	0x000000010d2afa8c in rstudio::session::modules::environment::(anonymous namespace)::callFramesAsJson(rstudio::session::modules::environment::(anonymous namespace)::LineDebugState*) at /Users/jmcphers/git/rstudio/src/cpp/session/modules/environment/SessionEnvironment.cpp:309
#37	0x000000010d2a78e3 in rstudio::session::modules::environment::(anonymous namespace)::commonEnvironmentStateData(int, bool, rstudio::session::modules::environment::(anonymous namespace)::LineDebugState*) at /Users/jmcphers/git/rstudio/src/cpp/session/modules/environment/SessionEnvironment.cpp:600
#38	0x000000010d2b716b in rstudio::session::modules::environment::(anonymous namespace)::enqueContextDepthChangedEvent(int, rstudio::session::modules::environment::(anonymous namespace)::LineDebugState*) at /Users/jmcphers/git/rstudio/src/cpp/session/modules/environment/SessionEnvironment.cpp:686
#39	0x000000010d2aae83 in rstudio::session::modules::environment::(anonymous namespace)::onConsolePrompt(rstudio_boost::shared_ptr<int>, rstudio_boost::shared_ptr<rstudio::session::modules::environment::(anonymous namespace)::LineDebugState>, rstudio_boost::shared_ptr<bool>, rstudio_boost::shared_ptr<rstudio::r::context::RCntxt>) at /Users/jmcphers/git/rstudio/src/cpp/session/modules/environment/SessionEnvironment.cpp:801
#40	0x000000010d2c28e2 in void rstudio_boost::_bi::list4<rstudio_boost::_bi::value<rstudio_boost::shared_ptr<int> >, rstudio_boost::_bi::value<rstudio_boost::shared_ptr<rstudio::session::modules::environment::(anonymous namespace)::LineDebugState> >, rstudio_boost::_bi::value<rstudio_boost::shared_ptr<bool> >, rstudio_boost::_bi::value<rstudio_boost::shared_ptr<rstudio::r::context::RCntxt> > >::operator()<void (*)(rstudio_boost::shared_ptr<int>, rstudio_boost::shared_ptr<rstudio::session::modules::environment::(anonymous namespace)::LineDebugState>, rstudio_boost::shared_ptr<bool>, rstudio_boost::shared_ptr<rstudio::r::context::RCntxt>), rstudio_boost::_bi::rrlist1<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&> >(rstudio_boost::_bi::type<void>, void (*&)(rstudio_boost::shared_ptr<int>, rstudio_boost::shared_ptr<rstudio::session::modules::environment::(anonymous namespace)::LineDebugState>, rstudio_boost::shared_ptr<bool>, rstudio_boost::shared_ptr<rstudio::r::context::RCntxt>), rstudio_boost::_bi::rrlist1<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&>&, int) at /opt/rstudio-tools/boost/boost_1_69_0/include/boost/bind/bind.hpp:463
#41	0x000000010d2c27cd in void rstudio_boost::_bi::bind_t<void, void (*)(rstudio_boost::shared_ptr<int>, rstudio_boost::shared_ptr<rstudio::session::modules::environment::(anonymous namespace)::LineDebugState>, rstudio_boost::shared_ptr<bool>, rstudio_boost::shared_ptr<rstudio::r::context::RCntxt>), rstudio_boost::_bi::list4<rstudio_boost::_bi::value<rstudio_boost::shared_ptr<int> >, rstudio_boost::_bi::value<rstudio_boost::shared_ptr<rstudio::session::modules::environment::(anonymous namespace)::LineDebugState> >, rstudio_boost::_bi::value<rstudio_boost::shared_ptr<bool> >, rstudio_boost::_bi::value<rstudio_boost::shared_ptr<rstudio::r::context::RCntxt> > > >::operator()<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&>(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) at /opt/rstudio-tools/boost/boost_1_69_0/include/boost/bind/bind.hpp:1306
#42	0x000000010d2c24f8 in rstudio_boost::detail::function::void_function_obj_invoker1<rstudio_boost::_bi::bind_t<void, void (*)(rstudio_boost::shared_ptr<int>, rstudio_boost::shared_ptr<rstudio::session::modules::environment::(anonymous namespace)::LineDebugState>, rstudio_boost::shared_ptr<bool>, rstudio_boost::shared_ptr<rstudio::r::context::RCntxt>), rstudio_boost::_bi::list4<rstudio_boost::_bi::value<rstudio_boost::shared_ptr<int> >, rstudio_boost::_bi::value<rstudio_boost::shared_ptr<rstudio::session::modules::environment::(anonymous namespace)::LineDebugState> >, rstudio_boost::_bi::value<rstudio_boost::shared_ptr<bool> >, rstudio_boost::_bi::value<rstudio_boost::shared_ptr<rstudio::r::context::RCntxt> > > >, void, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&>::invoke(rstudio_boost::detail::function::function_buffer&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) at /opt/rstudio-tools/boost/boost_1_69_0/include/boost/function/function_template.hpp:158
#43	0x000000010c5efa7f in rstudio_boost::function1<void, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&>::operator()(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) const at /opt/rstudio-tools/boost/boost_1_69_0/include/boost/function/function_template.hpp:763
#44	0x000000010c5ef9f5 in rstudio_boost::signals2::detail::void_type rstudio_boost::signals2::detail::call_with_tuple_args<rstudio_boost::signals2::detail::void_type>::m_invoke<rstudio_boost::function<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)>, 0u, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&>(rstudio_boost::function<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)>&, rstudio_boost::signals2::detail::unsigned_meta_array<0u>, std::__1::tuple<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&> const&, rstudio_boost::enable_if<rstudio_boost::is_void<rstudio_boost::function<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)>::result_type>, void>::type*) const at /opt/rstudio-tools/boost/boost_1_69_0/include/boost/signals2/detail/variadic_slot_invoker.hpp:105
#45	0x000000010c5ef999 in rstudio_boost::signals2::detail::void_type rstudio_boost::signals2::detail::call_with_tuple_args<rstudio_boost::signals2::detail::void_type>::operator()<rstudio_boost::function<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)>, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, 1ul>(rstudio_boost::function<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)>&, std::__1::tuple<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&> const&, mpl_::size_t<1ul>) const at /opt/rstudio-tools/boost/boost_1_69_0/include/boost/signals2/detail/variadic_slot_invoker.hpp:90
#46	0x000000010c5ef961 in rstudio_boost::signals2::detail::void_type rstudio_boost::signals2::detail::variadic_slot_invoker<rstudio_boost::signals2::detail::void_type, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&>::operator()<rstudio_boost::shared_ptr<rstudio_boost::signals2::detail::connection_body<std::__1::pair<rstudio_boost::signals2::detail::slot_meta_group, rstudio_boost::optional<int> >, rstudio_boost::signals2::slot<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&), rstudio_boost::function<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)> >, rstudio_boost::signals2::mutex> > >(rstudio_boost::shared_ptr<rstudio_boost::signals2::detail::connection_body<std::__1::pair<rstudio_boost::signals2::detail::slot_meta_group, rstudio_boost::optional<int> >, rstudio_boost::signals2::slot<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&), rstudio_boost::function<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)> >, rstudio_boost::signals2::mutex> > const&) const at /opt/rstudio-tools/boost/boost_1_69_0/include/boost/signals2/detail/variadic_slot_invoker.hpp:133
#47	0x000000010c5ef855 in rstudio_boost::signals2::detail::slot_call_iterator_t<rstudio_boost::signals2::detail::variadic_slot_invoker<rstudio_boost::signals2::detail::void_type, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&>, std::__1::__list_iterator<rstudio_boost::shared_ptr<rstudio_boost::signals2::detail::connection_body<std::__1::pair<rstudio_boost::signals2::detail::slot_meta_group, rstudio_boost::optional<int> >, rstudio_boost::signals2::slot<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&), rstudio_boost::function<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)> >, rstudio_boost::signals2::mutex> >, void*>, rstudio_boost::signals2::detail::connection_body<std::__1::pair<rstudio_boost::signals2::detail::slot_meta_group, rstudio_boost::optional<int> >, rstudio_boost::signals2::slot<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&), rstudio_boost::function<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)> >, rstudio_boost::signals2::mutex> >::dereference() const at /opt/rstudio-tools/boost/boost_1_69_0/include/boost/signals2/detail/slot_call_iterator.hpp:110
#48	0x000000010c5ef7c5 in rstudio_boost::signals2::detail::slot_call_iterator_t<rstudio_boost::signals2::detail::variadic_slot_invoker<rstudio_boost::signals2::detail::void_type, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&>, std::__1::__list_iterator<rstudio_boost::shared_ptr<rstudio_boost::signals2::detail::connection_body<std::__1::pair<rstudio_boost::signals2::detail::slot_meta_group, rstudio_boost::optional<int> >, rstudio_boost::signals2::slot<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&), rstudio_boost::function<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)> >, rstudio_boost::signals2::mutex> >, void*>, rstudio_boost::signals2::detail::connection_body<std::__1::pair<rstudio_boost::signals2::detail::slot_meta_group, rstudio_boost::optional<int> >, rstudio_boost::signals2::slot<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&), rstudio_boost::function<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)> >, rstudio_boost::signals2::mutex> >::reference rstudio_boost::iterators::iterator_core_access::dereference<rstudio_boost::signals2::detail::slot_call_iterator_t<rstudio_boost::signals2::detail::variadic_slot_invoker<rstudio_boost::signals2::detail::void_type, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&>, std::__1::__list_iterator<rstudio_boost::shared_ptr<rstudio_boost::signals2::detail::connection_body<std::__1::pair<rstudio_boost::signals2::detail::slot_meta_group, rstudio_boost::optional<int> >, rstudio_boost::signals2::slot<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&), rstudio_boost::function<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)> >, rstudio_boost::signals2::mutex> >, void*>, rstudio_boost::signals2::detail::connection_body<std::__1::pair<rstudio_boost::signals2::detail::slot_meta_group, rstudio_boost::optional<int> >, rstudio_boost::signals2::slot<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&), rstudio_boost::function<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)> >, rstudio_boost::signals2::mutex> > >(rstudio_boost::signals2::detail::slot_call_iterator_t<rstudio_boost::signals2::detail::variadic_slot_invoker<rstudio_boost::signals2::detail::void_type, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&>, std::__1::__list_iterator<rstudio_boost::shared_ptr<rstudio_boost::signals2::detail::connection_body<std::__1::pair<rstudio_boost::signals2::detail::slot_meta_group, rstudio_boost::optional<int> >, rstudio_boost::signals2::slot<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&), rstudio_boost::function<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)> >, rstudio_boost::signals2::mutex> >, void*>, rstudio_boost::signals2::detail::connection_body<std::__1::pair<rstudio_boost::signals2::detail::slot_meta_group, rstudio_boost::optional<int> >, rstudio_boost::signals2::slot<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&), rstudio_boost::function<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)> >, rstudio_boost::signals2::mutex> > const&) at /opt/rstudio-tools/boost/boost_1_69_0/include/boost/iterator/iterator_facade.hpp:550
#49	0x000000010c5ef70d in rstudio_boost::iterators::detail::iterator_facade_base<rstudio_boost::signals2::detail::slot_call_iterator_t<rstudio_boost::signals2::detail::variadic_slot_invoker<rstudio_boost::signals2::detail::void_type, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&>, std::__1::__list_iterator<rstudio_boost::shared_ptr<rstudio_boost::signals2::detail::connection_body<std::__1::pair<rstudio_boost::signals2::detail::slot_meta_group, rstudio_boost::optional<int> >, rstudio_boost::signals2::slot<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&), rstudio_boost::function<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)> >, rstudio_boost::signals2::mutex> >, void*>, rstudio_boost::signals2::detail::connection_body<std::__1::pair<rstudio_boost::signals2::detail::slot_meta_group, rstudio_boost::optional<int> >, rstudio_boost::signals2::slot<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&), rstudio_boost::function<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)> >, rstudio_boost::signals2::mutex> >, rstudio_boost::signals2::detail::void_type, rstudio_boost::iterators::single_pass_traversal_tag, rstudio_boost::signals2::detail::void_type const&, long, false, false>::operator*() const at /opt/rstudio-tools/boost/boost_1_69_0/include/boost/iterator/iterator_facade.hpp:656
#50	0x000000010c5ef655 in void rstudio_boost::signals2::optional_last_value<void>::operator()<rstudio_boost::signals2::detail::slot_call_iterator_t<rstudio_boost::signals2::detail::variadic_slot_invoker<rstudio_boost::signals2::detail::void_type, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&>, std::__1::__list_iterator<rstudio_boost::shared_ptr<rstudio_boost::signals2::detail::connection_body<std::__1::pair<rstudio_boost::signals2::detail::slot_meta_group, rstudio_boost::optional<int> >, rstudio_boost::signals2::slot<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&), rstudio_boost::function<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)> >, rstudio_boost::signals2::mutex> >, void*>, rstudio_boost::signals2::detail::connection_body<std::__1::pair<rstudio_boost::signals2::detail::slot_meta_group, rstudio_boost::optional<int> >, rstudio_boost::signals2::slot<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&), rstudio_boost::function<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)> >, rstudio_boost::signals2::mutex> > >(rstudio_boost::signals2::detail::slot_call_iterator_t<rstudio_boost::signals2::detail::variadic_slot_invoker<rstudio_boost::signals2::detail::void_type, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&>, std::__1::__list_iterator<rstudio_boost::shared_ptr<rstudio_boost::signals2::detail::connection_body<std::__1::pair<rstudio_boost::signals2::detail::slot_meta_group, rstudio_boost::optional<int> >, rstudio_boost::signals2::slot<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&), rstudio_boost::function<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)> >, rstudio_boost::signals2::mutex> >, void*>, rstudio_boost::signals2::detail::connection_body<std::__1::pair<rstudio_boost::signals2::detail::slot_meta_group, rstudio_boost::optional<int> >, rstudio_boost::signals2::slot<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&), rstudio_boost::function<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)> >, rstudio_boost::signals2::mutex> >, rstudio_boost::signals2::detail::slot_call_iterator_t<rstudio_boost::signals2::detail::variadic_slot_invoker<rstudio_boost::signals2::detail::void_type, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&>, std::__1::__list_iterator<rstudio_boost::shared_ptr<rstudio_boost::signals2::detail::connection_body<std::__1::pair<rstudio_boost::signals2::detail::slot_meta_group, rstudio_boost::optional<int> >, rstudio_boost::signals2::slot<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&), rstudio_boost::function<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)> >, rstudio_boost::signals2::mutex> >, void*>, rstudio_boost::signals2::detail::connection_body<std::__1::pair<rstudio_boost::signals2::detail::slot_meta_group, rstudio_boost::optional<int> >, rstudio_boost::signals2::slot<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&), rstudio_boost::function<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)> >, rstudio_boost::signals2::mutex> >) const at /opt/rstudio-tools/boost/boost_1_69_0/include/boost/signals2/optional_last_value.hpp:57
#51	0x000000010c5eca1f in void rstudio_boost::signals2::detail::combiner_invoker<void>::operator()<rstudio_boost::signals2::optional_last_value<void>, rstudio_boost::signals2::detail::slot_call_iterator_t<rstudio_boost::signals2::detail::variadic_slot_invoker<rstudio_boost::signals2::detail::void_type, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&>, std::__1::__list_iterator<rstudio_boost::shared_ptr<rstudio_boost::signals2::detail::connection_body<std::__1::pair<rstudio_boost::signals2::detail::slot_meta_group, rstudio_boost::optional<int> >, rstudio_boost::signals2::slot<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&), rstudio_boost::function<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)> >, rstudio_boost::signals2::mutex> >, void*>, rstudio_boost::signals2::detail::connection_body<std::__1::pair<rstudio_boost::signals2::detail::slot_meta_group, rstudio_boost::optional<int> >, rstudio_boost::signals2::slot<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&), rstudio_boost::function<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)> >, rstudio_boost::signals2::mutex> > >(rstudio_boost::signals2::optional_last_value<void>&, rstudio_boost::signals2::detail::slot_call_iterator_t<rstudio_boost::signals2::detail::variadic_slot_invoker<rstudio_boost::signals2::detail::void_type, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&>, std::__1::__list_iterator<rstudio_boost::shared_ptr<rstudio_boost::signals2::detail::connection_body<std::__1::pair<rstudio_boost::signals2::detail::slot_meta_group, rstudio_boost::optional<int> >, rstudio_boost::signals2::slot<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&), rstudio_boost::function<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)> >, rstudio_boost::signals2::mutex> >, void*>, rstudio_boost::signals2::detail::connection_body<std::__1::pair<rstudio_boost::signals2::detail::slot_meta_group, rstudio_boost::optional<int> >, rstudio_boost::signals2::slot<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&), rstudio_boost::function<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)> >, rstudio_boost::signals2::mutex> >, rstudio_boost::signals2::detail::slot_call_iterator_t<rstudio_boost::signals2::detail::variadic_slot_invoker<rstudio_boost::signals2::detail::void_type, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&>, std::__1::__list_iterator<rstudio_boost::shared_ptr<rstudio_boost::signals2::detail::connection_body<std::__1::pair<rstudio_boost::signals2::detail::slot_meta_group, rstudio_boost::optional<int> >, rstudio_boost::signals2::slot<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&), rstudio_boost::function<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)> >, rstudio_boost::signals2::mutex> >, void*>, rstudio_boost::signals2::detail::connection_body<std::__1::pair<rstudio_boost::signals2::detail::slot_meta_group, rstudio_boost::optional<int> >, rstudio_boost::signals2::slot<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&), rstudio_boost::function<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)> >, rstudio_boost::signals2::mutex> >) const at /opt/rstudio-tools/boost/boost_1_69_0/include/boost/signals2/detail/result_type_wrapper.hpp:64
#52	0x000000010c5ec5b2 in rstudio_boost::signals2::detail::signal_impl<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&), rstudio_boost::signals2::optional_last_value<void>, int, std::__1::less<int>, rstudio_boost::function<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)>, rstudio_boost::function<void (rstudio_boost::signals2::connection const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)>, rstudio_boost::signals2::mutex>::operator()(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) at /opt/rstudio-tools/boost/boost_1_69_0/include/boost/signals2/detail/signal_template.hpp:242
#53	0x000000010c5ded0c in rstudio_boost::signals2::signal<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&), rstudio_boost::signals2::optional_last_value<void>, int, std::__1::less<int>, rstudio_boost::function<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)>, rstudio_boost::function<void (rstudio_boost::signals2::connection const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)>, rstudio_boost::signals2::mutex>::operator()(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) at /opt/rstudio-tools/boost/boost_1_69_0/include/boost/signals2/detail/signal_template.hpp:722
#54	0x000000010c5de04a in rstudio::session::console_input::(anonymous namespace)::consolePrompt(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, bool) at /Users/jmcphers/git/rstudio/src/cpp/session/SessionConsoleInput.cpp:89
#55	0x000000010c5de2d9 in rstudio::session::console_input::rConsoleRead(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, bool, rstudio::r::session::RConsoleInput*) at /Users/jmcphers/git/rstudio/src/cpp/session/SessionConsoleInput.cpp:190
#56	0x000000010c859e5c in rstudio_boost::detail::function::function_invoker3<bool (*)(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, bool, rstudio::r::session::RConsoleInput*), bool, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, bool, rstudio::r::session::RConsoleInput*>::invoke(rstudio_boost::detail::function::function_buffer&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, bool, rstudio::r::session::RConsoleInput*) at /opt/rstudio-tools/boost/boost_1_69_0/include/boost/function/function_template.hpp:100
#57	0x000000010e014656 in rstudio_boost::function3<bool, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, bool, rstudio::r::session::RConsoleInput*>::operator()(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, bool, rstudio::r::session::RConsoleInput*) const at /opt/rstudio-tools/boost/boost_1_69_0/include/boost/function/function_template.hpp:763
#58	0x000000010e013d2d in rstudio::r::session::RReadConsole(char const*, unsigned char*, int, int) at /Users/jmcphers/git/rstudio/src/cpp/r/session/RStdCallbacks.cpp:317
```

**Comment by jmcphers** [comment]
Yes, this looks new -- `R_bcstack_t *bcprottop` was added before `srcref`, so we are probably crashing because we treat it like a srcref.

https://github.com/wch/r-source/blob/a0220b61bc6c8bd7158262170e9990fc83fc22d4/src/include/Defn.h#L560-L561

This was added a couple of months ago:

https://github.com/wch/r-source/commit/321ca345988101e5bb392427baf03a90d6ca2a46

**Comment by rich-rstudio** [comment]
verified version 1.3.908

**Comment by m-signo** [comment]
Since I installed R 4.0.0, every now and then RStudio (Version 1.2.5001) crashes reporting as error:
`Error in (function (srcref) : unimplemented type (29) in 'eval'`
I am not sure what causes this error, as it occurs with different scripts involving very different tasks and it doesn't seem to be related to a single function.

**Comment by jmcphers** [comment]
> RStudio (Version 1.2.5001) 

That's the problem -- RStudio 1.2.5001 is not compatible with R 4.0. If you are using R 4.0, we recommend the newest preview of RStudio:

https://www.rstudio.com/products/rstudio/download/preview/

However our latest stable version (1.2.5042) also has a fix for the crashing bug with R 4.0.
